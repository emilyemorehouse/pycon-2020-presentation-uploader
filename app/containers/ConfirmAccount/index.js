export { default as ConfirmAccount } from './ConfirmAccount'
