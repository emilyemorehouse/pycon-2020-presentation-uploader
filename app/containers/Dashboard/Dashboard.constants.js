// If the size is small, we only see 1 column
// If the size is medium, we only see 2 columns
// If the size is either large or xlarge, we see 3 columns
export const GRID_COLUMNS = {
  small: ['100%'],
  medium: ['1/2', '1/2'],
  large: ['1/3', '1/3', '1/3'],
  xlarge: ['1/3', '1/3', '1/3'],
}

// If the size is small, we have 3 rows
// If the size is medium, we have 2 rows
// If the size is large or xlarge, we have 1 row
export const GRID_ROWS = {
  small: ['small', 'small', 'small'],
  medium: ['medium', 'medium'],
  large: ['large'],
  xlarge: ['large'],
}

// Set the different areas you need for every size, defined as [col, row]
export const GRID_AREAS = {
  small: [
    { name: 'shows', start: [0, 0], end: [1, 0] },
    { name: 'todos', start: [0, 1], end: [1, 1] },
    { name: 'artist-venues', start: [0, 2], end: [1, 2] },
  ],
  // Shows pops to the top with todos and artist-venues split below
  medium: [
    { name: 'shows', start: [0, 0], end: [1, 0] },
    { name: 'todos', start: [0, 1], end: [0, 1] },
    { name: 'artist-venues', start: [1, 1], end: [1, 1] },
  ],
  large: [
    { name: 'todos', start: [0, 0], end: [0, 0] },
    { name: 'shows', start: [1, 0], end: [1, 0] },
    { name: 'artist-venues', start: [2, 0], end: [2, 0] },
  ],
  xlarge: [
    { name: 'todos', start: [0, 0], end: [0, 0] },
    { name: 'shows', start: [1, 0], end: [1, 0] },
    { name: 'artist-venues', start: [2, 0], end: [2, 0] },
  ],
}
