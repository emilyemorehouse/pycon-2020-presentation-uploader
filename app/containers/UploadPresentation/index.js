export { default as UploadPresentation } from './UploadPresentation'
