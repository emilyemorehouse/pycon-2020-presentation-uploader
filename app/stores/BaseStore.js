import { forEach } from 'lodash'
import { action } from 'mobx'

export class BaseStore {
  /**
   * mobx-sync does not set the default values without interacting with at
   * least one of the class variables that are set up as an observable
   *
   * This takes the intial state provided to the class instantiation and runs
   * through each value from the `initialState` that must be set to ensure the
   * default values are correct
   *
   * @param {object} initialState
   */
  constructor(initialState = {}) {
    forEach(initialState, (value, key) => {
      this[key] = value
    })
    this._initialState = initialState
  }

  @action.bound
  reset() {
    forEach(this._initialState, (value, key) => {
      this[key] = value
    })
  }
}
