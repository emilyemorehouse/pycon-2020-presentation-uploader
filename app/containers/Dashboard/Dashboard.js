import React from 'react'
import { Header } from 'grommet' /** @todo replace with wrapper component */
import { observer } from 'mobx-react'

// Components
import { Box } from 'components/Box'
import { Button } from 'components/Button'
import { Grid } from 'components/Grid'
import { Heading } from 'components/Heading'
import { Text } from 'components/Text'

// Helpers
import { GRID_COLUMNS, GRID_ROWS, GRID_AREAS } from './Dashboard.constants'

/**
 *
 * Dashboard
 *
 */
const Dashboard = observer(() => (
  <>
    <Heading level="2" pad="small">
      Dashboard
    </Heading>

    <Grid
      areas={GRID_AREAS}
      columns={GRID_COLUMNS}
      fill="vertical"
      gap="small"
      pad="small"
      justifyContent="center"
      rows={GRID_ROWS}
    >
      <Box gridArea="shows">
        <Heading margin="small" level={4}>
          Shows
        </Heading>

        <Header background="light-5" pad="medium">
          <Text>Sort</Text>
          <Button primary label="Add New"></Button>
        </Header>

        <Box fill border={{ color: 'light-5', size: 'thin' }}></Box>
      </Box>

      <Box gridArea="todos">
        <Heading margin="small" level={4}>
          To Dos
        </Heading>

        <Header background="light-5" pad="medium">
          <Text>Sort</Text>
          <Button primary label="Add New"></Button>
        </Header>

        <Box fill border={{ color: 'light-5', size: 'thin' }}></Box>
      </Box>

      <Box gridArea="artist-venues">
        <Heading margin="small" level={4}>
          Artists & Venues
        </Heading>

        <Header background="light-5" pad="medium">
          <Text>Sort</Text>
          <Button primary label="Add New"></Button>
        </Header>

        <Box fill border={{ color: 'light-5', size: 'thin' }}></Box>
      </Box>
    </Grid>
  </>
))

export default Dashboard
