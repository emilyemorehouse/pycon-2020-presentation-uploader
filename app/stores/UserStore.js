import { action, computed, observable } from 'mobx'
import { ignore } from 'mobx-sync'
import { createContext } from 'react'

import { BaseStore } from './BaseStore'

export const INITIAL_STATE = {
  accessToken: '',
  error: '',
  isLoading: false,
  message: '',
  refreshToken: '',
  user: {},
}

export class UserStore extends BaseStore {
  @observable accessToken = INITIAL_STATE.accessToken

  @ignore
  @observable
  error = INITIAL_STATE.error

  @ignore
  @observable
  isLoading = INITIAL_STATE.isLoading

  @ignore
  @observable
  message = INITIAL_STATE.message

  @observable refreshToken = INITIAL_STATE.refreshToken

  @observable user = INITIAL_STATE.user

  @computed
  get isAuthenticated() {
    return this.accessToken !== ''
  }

  @action.bound
  setCurrentUser({ accessToken = '', refreshToken = '', ...user } = {}) {
    this.accessToken = accessToken
    this.refreshToken = refreshToken
    this.user = {
      ...user,
    }
  }
}

export const user = new UserStore(INITIAL_STATE)
export const UserStoreContext = createContext(user)
