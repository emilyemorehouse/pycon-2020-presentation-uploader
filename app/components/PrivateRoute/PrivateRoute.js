import React, { useContext } from 'react'
import { Route, Redirect } from 'react-router-dom'
import { observer } from 'mobx-react'

// Stores
import { UserStoreContext } from '../../stores/UserStore'

/**
 * PrivateRoute
 */
const PrivateRoute = observer(({ component: Component, roles, ...rest }) => {
  const { isAuthenticated } = useContext(UserStoreContext)

  // User does not have an auth token - redirect to Login
  if (!isAuthenticated) {
    return (
      <Route
        {...rest}
        render={props => <Redirect to={{ pathname: '/login', state: { from: props.location } }} />}
      />
    )
  }

  // Base case - render the requested component
  return <Route {...rest} render={props => <Component {...props} />} />
})

export default PrivateRoute
