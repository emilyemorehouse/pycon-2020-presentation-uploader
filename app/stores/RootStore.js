/** @todo Fix import cycle */
/* eslint-disable import/no-cycle */

import { ignore } from 'mobx-sync'
import { action } from 'mobx'
import { createContext } from 'react'

import { locale } from './LocaleStore'
import { user } from './UserStore'
import { refreshAccessToken as refreshAccessTokenService } from '../services/user.service'

export class RootStore {
  @ignore storeLoaded = false

  locale = locale

  user = user

  @action.bound
  clearStore() {
    this.user.reset()
  }

  @action.bound
  async triggerRefreshToken(failedRequest) {
    try {
      const { access } = await refreshAccessTokenService(this.refreshToken)
      this.user.accessToken = access

      // Reassign to `failedRequest` as this object is used by the token
      // interceptor to retry the request
      // eslint-disable-next-line no-param-reassign
      failedRequest.response.config.headers.Authorization = `Bearer ${access}`
    } catch (err) {
      this.clearStore()
      throw err
    }
  }
}

export const store = new RootStore()
export const RootStoreContext = createContext(store)
