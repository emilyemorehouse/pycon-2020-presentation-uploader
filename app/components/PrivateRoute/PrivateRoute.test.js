import React from 'react'
import { IntlProvider } from 'react-intl'
import { MemoryRouter } from 'react-router-dom'
import { act, render, waitForElement } from '@testing-library/react'
import { Grommet, grommet } from 'grommet'
import '@testing-library/jest-dom/extend-expect'

import { Dashboard } from 'containers/Dashboard'
import trunk from '../../configureStore'
import { DEFAULT_LOCALE } from '../../i18n'
import { UserStore, UserStoreContext } from '../../stores/UserStore'
import PrivateRoute from './PrivateRoute'

const TEST_USER = 'suki@cuttlesoft.com'

describe('<PrivateRoute />', () => {
  let userStore = null

  beforeAll(() => {
    userStore = new UserStore()
    trunk.init()
  })

  const renderComponent = userStoreValue =>
    render(
      <Grommet theme={grommet}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <MemoryRouter initialEntries={['/dashboard']}>
            <UserStoreContext.Provider value={userStoreValue}>
              <PrivateRoute path="/dashboard" component={Dashboard} />
            </UserStoreContext.Provider>
          </MemoryRouter>
        </IntlProvider>
      </Grommet>,
    )

  it('does not render if user is not authenticated', async () => {
    const { container } = renderComponent(userStore)

    expect(container).toMatchSnapshot()
    expect(container).not.toHaveTextContent('Dashboard')
  })

  it('renders dashboard if user is authenticated', async () => {
    userStore.accessToken = 'faketoken'
    userStore.email = TEST_USER

    act(async () => {
      const { container, getByTestId } = renderComponent(userStore)
      const dashboardPage = await waitForElement(() => getByTestId('dashboard-page'))

      expect(container).toMatchSnapshot()
      expect(dashboardPage).toHaveTextContent('Dashboard')
    })
  })
})
