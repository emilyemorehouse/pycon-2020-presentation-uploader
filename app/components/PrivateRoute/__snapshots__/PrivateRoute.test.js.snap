// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrivateRoute /> does not render if user is not authenticated 1`] = `
<div>
  <div
    class="StyledGrommet-sc-19lkkz7-0 gVQrIe"
  />
</div>
`;
